vowpalwabbit/core/tests/io_alignment_test.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw/core/io_buf.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+using align_t = VW::desired_align::align_t;
+using flatbuffer_t = VW::desired_align::flatbuffer_t;
+using VW::desired_align;
+
+struct positioned_ptr
+{
+  size_t allocation;
```

Comment:
```suggestion
  /* allocation holds the size of the memory allocation */
```suggestion
  size_t allocation;
```